crates/ruff_linter/src/rules/flake8_quotes/rules/check_string_quotes.rs

```diff
@@ -362,120 +351,34 @@ fn strings(
                 fixed_contents,
                 *range,
             )));
-            diagnostics.push(diagnostic);
-        }
-    }
-
-    diagnostics
-}
-
-/// A builder for the f-string range.
-///
-/// For now, this is limited to the outermost f-string and doesn't support
-/// nested f-strings.
-#[derive(Debug, Default)]
-struct FStringRangeBuilder {
-    start_location: TextSize,
-    end_location: TextSize,
-    nesting: u32,
-}
-
-impl FStringRangeBuilder {
-    fn visit_token(&mut self, token: &Tok, range: TextRange) {
-        match token {
-            Tok::FStringStart(_) => {
-                if self.nesting == 0 {
-                    self.start_location = range.start();
-                }
-                self.nesting += 1;
-            }
-            Tok::FStringEnd => {
-                self.nesting = self.nesting.saturating_sub(1);
-                if self.nesting == 0 {
-                    self.end_location = range.end();
-                }
-            }
-            _ => {}
+            checker.diagnostics.push(diagnostic);
         }
     }
-
-    /// Returns `true` if the lexer is currently inside of a f-string.
-    ///
-    /// It'll return `false` once the `FStringEnd` token for the outermost
-    /// f-string is visited.
-    const fn in_fstring(&self) -> bool {
-        self.nesting > 0
-    }
-
-    /// Returns the complete range of the previously visited f-string.
-    ///
-    /// This method should only be called once the lexer is outside of any
-    /// f-string otherwise it might return an invalid range.
-    ///
-    /// It doesn't consume the builder because there can be multiple f-strings
-    /// throughout the source code.
-    fn finish(&self) -> TextRange {
-        debug_assert!(!self.in_fstring());
-        TextRange::new(self.start_location, self.end_location)
-    }
 }
 
 /// Generate `flake8-quote` diagnostics from a token stream.
-pub(crate) fn check_string_quotes(
-    diagnostics: &mut Vec<Diagnostic>,
-    lxr: &[LexResult],
-    locator: &Locator,
-    settings: &LinterSettings,
-) {
-    // Keep track of sequences of strings, which represent implicit string
-    // concatenation, and should thus be handled as a single unit.
-    let mut sequence = vec![];
-    let mut state_machine = StateMachine::default();
-    let mut fstring_range_builder = FStringRangeBuilder::default();
-    for &(ref tok, range) in lxr.iter().flatten() {
-        fstring_range_builder.visit_token(tok, range);
-        if fstring_range_builder.in_fstring() {
-            continue;
-        }
+pub(crate) fn check_string_quotes(checker: &mut Checker, string_like: StringLike) {
+    // If the string is part of a f-string, ignore it.
+    if checker
+        .indexer()
+        .fstring_ranges()
+        .outermost(string_like.start())
+        .is_some_and(|outer_range| outer_range != string_like.range())
+    {
+        return;
+    }
```

Comment:
TODO: I just realized we could probably just use `checker.semantic().in_fstring()`

Reply:
Actually, this will be true even for the _outermost_ f-string.